backend/lucene/src/main/java/org/hibernate/search/backend/lucene/orchestration/impl/LuceneParallelWorkOrchestratorImpl.java

```diff
@@ -60,7 +60,7 @@ protected void doStart(ConfigurationPropertySource propertySource) {
 
 	@Override
 	protected void doSubmit(WorkExecution<?> workExecution, OperationSubmitter operationSubmitter) throws InterruptedException {
-		operationSubmitter.submitToExecutor( executor, workExecution, blockingRetryProducer );
+		operationSubmitter.submitToExecutor( executor, workExecution, blockingRetryProducer, workExecution::markAsFailed );
```

Comment:
Use a BiConsumer to avoid capturing lambdas, which have a performance impact:

```suggestion
		operationSubmitter.submitToExecutor( executor, workExecution, blockingRetryProducer, WorkExecution::markAsFailed );
```

Also, you could put `WorkExecution::markAsFailed` into a `static final` constant.

Reply:
> Use a BiConsumer to avoid capturing lambdas, which have a performance impact:

Thanks, I knew there was something behind it 😃